HW0/P6/P6.txt

```diff
@@ -1 +1,16 @@
 # Your answers here
+Part A: 
+
+Everytime we run this code, we get that the threads take slightly different times. 
+This is true because we expect that all 4 threads finish at the same time, so then 
+4 new items will be executed. However, we see that some of these 4 threads, even 
+though they all should take 0.25 seconds, finish just a tad faster than others. This 
+means that we should not use waiting times to try to make code synchronous. For example, 
+if we tried to run processes A1, A2, A3 and B1, B2, B3, where all the A's run before the
+B's. We should not just put them into a list in that order because we should make no 
+time guarantees that all the A's will finish before any B's will. 
```

Comment:
The gist of your answer seems the same to what I have written. A few use cases where this might fail includes serial execution. Lets say we want to process a lengthy time series where the time values are not independent and are functions of the events that have previously occurred. Also if we think of the Kalman filtering process, we won't be able to run in on the multiprocessing module since the updates depend on whether the previous updates have been executed.